### ISSUE TITLE
external UI: no keyboard focus with window embedding

### ISSUE THREAD

**Issue by jpcima** [DISTRHO/DPF#147]
Hi, this example branch modifies `ExternalExampleUI`, in order to embed tk with `wish` interpreter as a subprocess. For this purpose, the `EMBED_UI` has been enabled.

https://github.com/jpcima/DPF/tree/tk

Problem: when it's embedded, it does not receive the keyboard focus.
In the example, there is a central text box which is configured to be editable.

In a JACK standalone, it is editable. In carla-single or jalv LV2, it isn't.
It has to do with the embedding: when `getNextWindowId()` is overriden with a value `0` to disable, the window is separate and I am able type text in the LV2.

**Comment by jpcima** [comment]
update: as it seems, it's specifically a Tk problem.
I tried embedding another process supporting it (mpv with --wid flag), and in this case keyboard worked.

I'll contribute a Tk-based example if I manage..

Other thing: external UI does not have the uiIdle callback, can it get one?
It will be useful to communicate in ExternalProcess->UI direction.
(the case of checking a FIFO IPC and posting state change to DSP, for example)

**Comment by jpcima** [comment]
This case of keyboard focus is not a problem a DPF, I close it.
